server/src/main/java/com/cloud/user/AccountManagerImpl.java

```diff
@@ -1712,25 +1720,48 @@ public AccountVO updateAccount(UpdateAccountCmd cmd) {
         }
 
         // Check if user performing the action is allowed to modify this account
-        checkAccess(getCurrentCallingAccount(), _domainMgr.getDomain(account.getDomainId()));
+        Account caller = getCurrentCallingAccount();
+        checkAccess(caller, _domainMgr.getDomain(account.getDomainId()));
 
-        // check if the given account name is unique in this domain for updating
-        Account duplicateAcccount = _accountDao.findActiveAccount(newAccountName, domainId);
-        if (duplicateAcccount != null && duplicateAcccount.getId() != account.getId()) {
-            throw new InvalidParameterValueException(
-                    "There already exists an account with the name:" + newAccountName + " in the domain:" + domainId + " with existing account id:" + duplicateAcccount.getId());
+        if(newAccountName != null) {
+
+            if (newAccountName.isEmpty()) {
```

Comment:
@bwsw could you please add a comment here to specify that you are checking that the user did not pass an empty string for an optional parameter. Otherwise I think it is a bit confusing why you check if a not null string is empty. At least I found it confusing until I saw the conversation between you and @rafaelweingartner. Overall LGTM

Reply:
@bwsw can you please address @dhlaluku comment? Thanks!